covid19-dashboard/main.py

```diff
@@ -11,181 +11,110 @@
+@app.route("/get-places")
+def get_places():
+    """Returns a dictionary of dcid->(name, type)
+    where name is the name of the region and type is the type of region."""
+    response: dict = send_request(req_url='https://api.datacommons.org/node/property-values',
+                                  req_json={"dcids": ['NYT_COVID19_County', 'NYT_COVID19_State'],
+                                            "property": "member"},
+                                  post=True,
+                                  compress=False,
+                                  api_key=API_KEY)
+
+    counties: dict = {county['dcid']: (county['name'], 'county') for county in response['NYT_COVID19_County']['out']}
+    states: dict = {state['dcid']: (state['name'], 'state') for state in response['NYT_COVID19_State']['out']}
+    return {**counties, **states}
+
+
+@cache.cached(timeout=50000)
+@app.route("/get-total-cases")
+def get_total_cases():
+    """Returns the total Cumulative Cases of each region per day. day->dcid->cases"""
+    dcids: list = list(get_places().keys())
+    return get_covid_data(dcids, 'NYTCovid19CumulativeCases')
+
+
+@cache.cached(timeout=50000)
+@app.route("/get-total-deaths")
+def get_total_deaths():
+    """Returns the total Cumulative Deaths of each region. day->dcid->deaths"""
+    dcids: list = list(get_places().keys())
+    return get_covid_data(dcids, 'NYTCovid19CumulativeDeaths')
+
+
+@cache.cached(timeout=50000)
+@app.route("/get-population")
+def get_population():
+    """Returns the total population of each region. dcid->population"""
+    dcids: list = list(get_places().keys())
+    response: dict = send_request("https://api.datacommons.org/bulk/stats",
+                                  {"place": dcids,
+                                   "stats_var": "TotalPopulation"},
+                                  api_key=API_KEY)
+    output: defaultdict = defaultdict()
+    for dcid in response:
+        if not response[dcid]:
+            continue
+        output[dcid] = response[dcid]['data']['2018']
+    return output
+
+
+def get_covid_data(place: list, stats_var: str) -> dict:
+    """
+    In charge of querying and returning the covid data in the appropriate format.
+    :param place: list of dcids to get data for
+    :param stats_var: NYTCovid19CumulativeCases or NYTCovid19CumulativeDeaths
+    :return: dictionary where every day contains several regions, and those regions contain an int of cases/deaths
+    """
+    response: dict = send_request("https://api.datacommons.org/bulk/stats",
+                                  {"place": place,
+                                   "stats_var": stats_var},
+                                  api_key=API_KEY)
+    output: defaultdict = defaultdict(dict)
+
+    # Store the data in output as dates->dcid->value
+    for dcid in response:
+        if "data" not in response[dcid]:
+            continue
+        data = response[dcid]["data"]
+        for day in data:
```

Comment:
can use date here, we could have non-day data later

Reply:
Fixed.